documentation/decisions/adr/2023-12-15-deliverable-reporting-strategy.md

```diff
@@ -0,0 +1,235 @@
+  - If stakeholders aren't familiar with the relationship between deliverable status and reporting, it may be harder for them to understand why certain deliverables appear in a given report but others do not.
+
+## Evaluation - Assigning issues to deliverables <!-- OPTIONAL -->
+
+### Milestone
+
+This option involves creating a milestone for each 30k deliverable and assigning issues to that milestone if they are required for delivery of that 30k.
+
+> [!NOTE]
+> This is a slightly different use of milestones than the project currently supports. Right now, the engineering team can create multiple milestones for a given 30k deliverable and reporting is driven by tagging the parent 30k deliverable in the description of the milestone. This new approach would require a *single* milestone per 30k deliverable and all tickets associated with that deliverable would need to be added to the milestone to be included in reporting.
+
+> [!TIP]
+> **Bottom line:** This is the best option if we:
+> - want a consistent way to filter and group issues by deliverable across all GitHub projects and reporting as well as within the repository
+> - but are okay with preventing the engineering team from using milestones to organize issues into smaller units of work
+
+
+- **Pros**
+  - Ensures that there is just one source of truth for the list of issues associated with a given 30k deliverable.
+  - Supports filtering both within the repository and within GitHub projects (e.g. roadmap and sprint board).
+  - Supports grouping both within the GitHub projects and within GitHub insight reporting.
+- **Cons**
+  - Prevents the engineering team from using GitHub milestones to organize and group related issues outside of the 30k deliverable framework.
+  - Milestones that represent 30k deliverables will have a *lot* of tickets unless we work on downscoping 30k deliverables.
+  - Does not support customizing the order of the groups on the sprint board when grouping by milestone -- milestones can only be sorted alphabetically.
+
+### Label
+
+This option involves creating a label for each 30k deliverable with a consistent prefix (e.g. `30k: Public launch`) and then applying this label to each of the issues required for delivery of that 30k.
+
+> [!TIP]
+> **Bottom line:** This is the best option if:
+> - we want a consistent way to filter issues by deliverable in the repository and across GitHub projects
+> - but can compromise on being able to **group** by deliverable in GitHub projects or reporting
+
+- **Pros**
+  - Allows the engineering team to continue to use GitHub milestones to organize issues into units of work that make sense to them.
+  - Supports filtering both within the repository and within GitHub projects (e.g. roadmap and sprint board).
+- **Cons**
+  - Does not easily support grouping in GitHub projects or GitHub insights reporting.
+  - Will result in a large number of labels if 30k labels aren't deleted once a 30k deliverable is complete.
```

Comment:
Similar to the cons I suggested above, my main concern is the work it takes to maintain multiple sources of truth. So if you change the name of a milestone or create a new one, you'd also need to create a new label. Not a big deal, and (I think) it's easy to rename labels and have the renamed value apply to every ticket. But probably worth noting.

Reply:
I added the steps involved in creating and renaming deliverables here: [feat: Adds steps and examples to reporting ADR](https://github.com/HHS/simpler-grants-gov/pull/926/commits/e21130f9b175611f0ea77576b2794391146b35b9)

I also made note of the number of steps required in the comparison table: [feat: Adds comparison table and links to examples](https://github.com/HHS/simpler-grants-gov/pull/926/commits/ae880e047deedc5f5cf014b99a67a20b753eb392)